bot/connector-google-chat/src/main/kotlin/GoogleChatConnectorCallback.kt

```diff
@@ -18,9 +18,66 @@ package ai.tock.bot.connector.googlechat
 
 import ai.tock.bot.connector.ConnectorCallbackBase
 import ai.tock.bot.connector.googlechat.builder.googleChatConnectorType
+import ai.tock.bot.engine.user.UserTimeline
+import com.google.api.services.chat.v1.HangoutsChat
+import com.google.api.services.chat.v1.model.Thread
+import mu.KotlinLogging
 
 data class GoogleChatConnectorCallback(
     override val applicationId: String,
     val spaceName: String,
     val threadName: String,
-) : ConnectorCallbackBase(applicationId, googleChatConnectorType)
+    private val chatService: HangoutsChat,
+    private val introMessage: String?,
+) : ConnectorCallbackBase(applicationId, googleChatConnectorType) {
+    private val logger = KotlinLogging.logger {}
+
+    /**
+     * Called when the UserTimeline is loaded.
+     * Sends the intro message if this is a new conversation (empty dialog).
+     */
+    override fun initialUserTimelineLoaded(userTimeline: UserTimeline) {
+        if (shouldSendIntro(userTimeline)) {
+            sendIntroMessage()
+        }
+    }
+
+    /**
+     * Determines if the intro message should be sent.
+     * Returns true if:
+     * - An intro message is configured
+     * - The user timeline has no bot actions (new conversation)
+     */
+    private fun shouldSendIntro(userTimeline: UserTimeline): Boolean = introMessage != null && !userTimeline.containsBotAction()
+
+    /**
+     * Sends the intro message to the Google Chat space/thread.
+     */
+    private fun sendIntroMessage() {
+        if (introMessage == null) return
+
+        try {
+            logger.info {
+                "Sending intro message to Google Chat: space=$spaceName, thread=$threadName"
+            }
+
+            val response =
+                chatService
+                    .spaces()
+                    .messages()
+                    .create(
+                        spaceName,
+                        GoogleChatConnectorTextMessageOut(introMessage)
+                            .toGoogleMessage()
+                            .setThread(Thread().setName(threadName)),
+                    ).setMessageReplyOption("REPLY_MESSAGE_FALLBACK_TO_NEW_THREAD")
+                    .execute()
```

Comment:
Did you check if the message is correctly logged into the user's dialog / timeline ? and then visible into the studio dialogs ?